### PR TITLE
[GCP] Use the latest debian image to align python 3.10

### LINKED ISSUE #1604: Python version mismatch when migrating from AWS to GCP
# Description of Problem
I recently migrated my workload from AWS to GCP and ran into issues with python versions. AWS's deep learning AMI is based on LTS versions of Ubuntu with Python 3.8+, whereas the default SkyPilot GCP image uses Debian 10 with Python 3.7. This resulted in an error as some libraries in my application relied on features from Python 3.8+. 

Note that the default SkyPilot image for AWS is based on Ubuntu 22.04 and uses Python 3.9.

I believe this is a SkyPilot issue because running workloads on any cloud requires a consistent environment. This can be addressed by launching SkyPilot tasks in images that are uniform across clouds with tools such as docker or the cloud's image marketplace.

# Current workaround
I've currently made some changes to my YAML files to set up and activate conda environments for the specific Python version I need:
```yaml
setup: |
  # Create an environment with Python 3.8.
  conda create -y -n my_task python=3.8
  source activate my_task
  # Install required Python packages.

run: |
  source activate experiments
  # Run the workload script.
```

Note that this workaround only addresses Python version mismatches. Version mismatches in other libraries and programs (e.g. `gcc`) will require a different solution. 

Posted at the request of @romilbhardwaj.

### PR DESCRIPTION
**Pull request by Michaelvll**
Fixes #1604 

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below)
  - [x] `sky launch -c test-new-debian --cloud gcp --cpus 2+`
  - [x] `sky launch -c test-new-debian-gpu --clou gcp --gpus T4 nvidia-smi`
- [x] All smoke tests: `pytest tests/test_smoke.py`  (except for #2309)
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: sky/clouds/service_catalog/gcp_catalog.py [hunk 1]
```diff
@@ -27,6 +27,11 @@
 _image_df = common.read_catalog('gcp/images.csv',
                                 pull_frequency_hours=_PULL_FREQUENCY_HOURS)
 
+if _image_df[_image_df['Tag'] == 'skypilot:gpu-debian-11'].empty:
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/gcp_catalog.py:30]: Q: do we expect doing this for future image tags? Or can we have some auto update mechanism in the future?

Comment (Michaelvll) [reply sky/clouds/service_catalog/gcp_catalog.py]: Good point! I changed it to automatically refresh when the tag is failed to find in the catalog. Wdyt?

File: sky/clouds/service_catalog/gcp_catalog.py [hunk 2]
```diff
@@ -551,11 +554,21 @@ def check_accelerator_attachable_to_host(instance_type: str,
 
 def get_image_id_from_tag(tag: str, region: Optional[str]) -> Optional[str]:
     """Returns the image id from the tag."""
-    return common.get_image_id_from_tag_impl(_image_df, tag, region)
+    global _image_df
+
+    image_id = common.get_image_id_from_tag_impl(_image_df, tag, region)
+    if image_id is None:
```
Comment (concretevitamin) [inline sky/clouds/service_catalog/gcp_catalog.py:560]: Is it possible that the caller passed in an image tag that doesn't exist, and the current impl will keep re-pulling?

**Review by concretevitamin — APPROVED** [review]
Thanks @Michaelvll - a question about refresh.

File: sky/clouds/service_catalog/gcp_catalog.py (hunk 2 above)
Comment (Michaelvll) [reply sky/clouds/service_catalog/gcp_catalog.py:560]: This should only re-pull once, when the image tag is not found, which seems to be fine as for the failure case, it is ok to have some overheads. Wdyt?

Comment (concretevitamin) [reply sky/clouds/service_catalog/gcp_catalog.py:560]: LGTM.

**Pull request merged**